plugins/chat/app/services/chat/action/calculate_memberships_for_removal.rb

```diff
@@ -59,12 +59,12 @@ def self.call(scoped_users:, channel_ids: nil)
         scoped_memberships =
           Chat::UserChatChannelMembership
             .joins(:chat_channel)
-            .where(user: scoped_users)
+            .where(user_id: scoped_users_query.select(:id))
             .where(chat_channel_id: channel_permissions_map.map(&:channel_id))
 
         memberships_to_remove = []
+
         scoped_memberships.find_each do |membership|
-          scoped_user = scoped_users.find { |su| su.id == membership.user_id }
```

Comment:
Moved this down to line 95 where we will execute the query and create the AR object only when we absolutely need to.